plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,658 @@
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    object_retention_info = {}
+
+    try:
+        object_retention_info = connection.get_object_retention(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        pass
+
+    if len(object_retention_info) != 0:
+        # Remove ResponseMetadata from object_retention_info, convert to snake_case
+        del(object_retention_info['ResponseMetadata'])
+        object_retention_info = camel_dict_to_snake_dict(object_retention_info)
+
+    return object_retention_info
+
+
+def describe_s3_object_tagging(connection, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    object_tagging_info = {}
+
+    try:
+        object_tagging_info = connection.get_object_tagging(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        pass
+
+    if len(object_tagging_info) != 0:
+        # Remove ResponseMetadata from object_tagging_info, convert to snake_case
+        del(object_tagging_info['ResponseMetadata'])
+        object_tagging_info = boto3_tag_list_to_ansible_dict(object_tagging_info['TagSet'])
+
+    return object_tagging_info
+
+
+def get_object_details(connection, module, bucket_name, object_name, requested_facts):
+
+    all_facts = {}
+
+    # Remove non-requested facts
+    requested_facts = {fact: value for fact, value in requested_facts.items() if value is True}
+
+    all_facts['object_data'] = get_object(connection, bucket_name, object_name)['object_data']
+
+    for key in requested_facts:
+        if key == 'object_acl':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_acl(connection, bucket_name, object_name)
+        elif key == 'object_attributes':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_attributes(connection, module, bucket_name, object_name)
+        elif key == 'object_legal_hold':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_legal_hold(connection, bucket_name, object_name)
+        elif key == 'object_lock_configuration':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_lock_configuration(connection, bucket_name)
+        elif key == 'object_retention':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_retention(connection, bucket_name, object_name)
+        elif key == 'object_tagging':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_tagging(connection, bucket_name, object_name)
+
+    return all_facts
+
+
+def get_object(connection, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    result = {}
+    object_info = {}
+
+    try:
+        object_info = connection.get_object(**params)
```

Comment:
This will download the actual object.  If the object is large this will have a big impact on performance.  client.head_object should provide the same keys without downloading the actual file.

Reply:
Fixed.